src/databricks/labs/ucx/install.py

```diff
@@ -282,18 +284,31 @@ def _confirm_force_install(self) -> bool:
         return False
 
     def configure(self, default_config: WorkspaceConfig | None = None) -> WorkspaceConfig:
+        connection_error_message = (
+            f"Cannot connect with {self.workspace_client.config.host} see "
+            "https://github.com/databrickslabs/ucx#network-connectivity-issues for help: %s"
+        )
         try:
             config = self.installation.load(WorkspaceConfig)
             self._compare_remote_local_versions()
-            if self._confirm_force_install():
-                return self._configure_new_installation(default_config)
-            self._apply_upgrades()
-            return config
+            if not self._confirm_force_install():
+                self._apply_upgrades()
+                return config
         except NotFound as err:
             logger.debug(f"Cannot find previous installation: {err}")
         except (PermissionDenied, SerdeError, ValueError, AttributeError):
             logger.warning(f"Existing installation at {self.installation.install_folder()} is corrupted. Skipping...")
-        return self._configure_new_installation(default_config)
+        except TimeoutError as err:
+            # API calls are wrapped with a retry mechanism for requests.exceptions.ConnectionError
+            # eventually raising a TimeoutError if tries take too long
+            if isinstance(err.__cause__, RequestsConnectionError):  # raise TimeoutError(...) from ConnectionError(...)
+                logger.warning(connection_error_message, err)
+        try:
+            return self._configure_new_installation(default_config)
+        except TimeoutError as err:
+            if isinstance(err.__cause__, RequestsConnectionError):
+                logger.warning(connection_error_message, err)
+            raise err
```

Comment:
I think the `TimeoutError` needs to be at a higher level than the existing catch blocks: if we hit the `TimeoutError` on the first call, I don't think it makes sense to continue the second call?

If we do want to continue I'd prefer (slightly) different messages:
 - The first message should indicate we're skipping.
 - Even though we normally have backtraces in the logs, it helps locate the right part of the code quickly when the messages are different. (In this case locating the installation versus configuring the new installation.

Reply:
> if we hit the TimeoutError on the first call, I don't think it makes sense to continue the second call?

If there is a `default_config` then we want to try

> I think the TimeoutError needs to be at a higher level than the existing catch blocks:

I thought so too at first, but it resulted in introducing it in a context where it not always made sense. 

It is not perfect, but my reasoning was that this is (hopefully) an exceptional flow. 99%+ of the cases a connection with Databricks can be made. Hence, my tradeoff of keeping code complexity lower vs that the backtraces are not right were the first try fails